codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -2540,12 +2552,41 @@ simulationControls w
     | playbackSpeed w == 0 = [PlayButton, StepButton, SpeedSlider]
     | otherwise = [PauseButton, SpeedSlider]
 
+debugSimulationControls :: Wrapped [w] -> [Control [w]]
+debugSimulationControls w
+    | lastInteractionTime w > 5 = []
+    | playbackSpeed w == 0 && not (null (tail (state w))) = [PlayButton, StepButton, SpeedSlider, UndoButton]
+    | playbackSpeed w == 0 = [PlayButton, StepButton, SpeedSlider]
+    | otherwise = [PauseButton, SpeedSlider]
+
 simulationOf simInitial simStep simDraw =
-    runPauseable initial simStep simulationControls simDraw
+    runInspect simulationControls initial simStep (\_ r -> r) simDraw
   where
     initial =
         Wrapped {state = simInitial, playbackSpeed = 1, lastInteractionTime = 1000, isDragging = False}
 
+debugSimulationOf simInitial simStep simDraw =
+    runInspect debugSimulationControls initial step (\_ r -> r) draw
+  where
+    initial =
+        Wrapped {state = [simInitial], playbackSpeed = 1, lastInteractionTime = 1000, isDragging = False}
+    step dt (x:xs) = simStep dt x : x : xs
```

Comment:
I wonder if you want to do the same thing here that you do in `debugInteractionOf`.  But it's not really important.  Simulations whose state doesn't change in `step` are very rare.

Reply:
I changed it to be similar to ```debugInteractionOf``` but didn't add the ```prepend``` function here. I can do that also if needed.